tests/js/server/paths/filesystem.js

```diff
@@ -0,0 +1,350 @@
+      assertEqual(raw_row,
+                  JSONParsed,
+                  'Expected ' + fn + ' to contain "' + JSONParsed + '" but it contained: "' + raw_row + '"!');
+      count ++;
+    });
+    assertEqual(count, 2);
+  }
+  function tryAdler32Allowed(fn, expectedContentPlain) {
+    let content = fs.adler32(fn);
+    assertTrue(content > 0);
+  }
+  function tryWriteAllowed(fn, text) {
+    let rc = fs.write(fn, text);
+    assertTrue(rc, 'Expected ' + fn + ' to be writeable');
+  }
+
+  function tryAppendAllowed(fn, text) {
+    let rc = fs.append(fn, text);
+    assertTrue(rc, 'Expected ' + fn + ' to be appendeable');
+  }
+
+  function tryChmodAllowed(fn) {
+    let rc = fs.chmod(fn, '0755');
+    assertTrue(rc, 'Expected ' + fn + ' to be chmodable');
+  }
+
+  function tryExistsAllowed(fn, exists) {
+    let rc = fs.exists(fn);
+    assertEqual(rc, exists, 'Expected ' + fn + ' to be stat-eable');
+    if (exists) {
+      rc = fs.mtime(fn);
+      assertTrue(rc > 0);
+    }
+  }
+
+  function tryFileSizeAllowed(fn) {
+    let rc = fs.size(fn);
+    assertTrue(rc, 'Expected ' + fn + ' to be stat-eable');
+  }
+
+  function tryIsDirectoryAllowed(fn, isDirectory) {
+    let rc = fs.isDirectory(fn);
+    assertEqual(rc, isDirectory, 'Expected ' + fn + ' to be a ' + isDirectory ? "directory.":"file.");
+  }
+
+  function tryCreateDirectoryAllowed(fn) {
+    let rc = fs.makeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryAllowed(fn) {
+    let rc = fs.removeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryCreateDirectoryRecursiveAllowed(fn) {
+    let rc = fs.makeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryRecursiveAllowed(fn) {
+    let rc = fs.removeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to be removed.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryIsFileAllowed(fn, isFile) {
+    let rc = fs.isFile(fn);
+    assertEqual(rc, isFile, 'Expected ' + fn + ' to be a ' + isFile ? "file.":"directory.");
+  }
+
+  function tryListFileAllowed(dn, expectCount) {
+    let rc = fs.list(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryListTreeAllowed(dn, expectCount) {
+    let rc = fs.listTree(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryGetTempFileAllowed(dn) {
+    let tfn = fs.getTempFile();
+    fs.write(tfn, "hello");
+    assertTrue(fs.isFile(tfn));
+    fs.remove(tfn);
+  }
+
+  function tryMakeAbsoluteAllowed(fn) {
+    fs.makeAbsolute(fn);
+  }
+  function tryCopyFileAllowed(sn, tn) {
+    try {
```

Comment:
this try...catch here is just unnecessary, and will obfuscate the stack trace by moving the error location from the `fs.copyFile()` to `assertTrue(false)`.